README.md

```diff
@@ -118,3 +101,171 @@ Remote debugging using :2000
+ * [ninja](https://ninja-build.org)
+
+Addtionally, depending on if you want to build/work on the firwmare or Black Magic Debug App (BMDA), you also need:
+
+#### Black Magic Debug Firmware
+
+ * [`arm-none-eabi-gcc`](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/downloads) (from ARM, self-built and distro-supplied is currently broken)
+
+If you wish to use the older [gnu-rm](https://developer.arm.com/downloads/-/gnu-rm) ARM toolchain, this is fine and works well.
+
+If you have a toolchain from other sources and find problems, check if it is an issue with your toolchain, if not open an issue or better provide a pull request with a fix.
+
+#### Black Magic Debug App
+
+ * [GCC](https://gcc.gnu.org) or Clang (Clang is not strictly officially supported)
+
+If you wish to enable support for 3rd party probes, and not only the official native hardware, you will also need:
+
+ * libusb
+ * libftdi
+ * evt
+ * libhidapi
+
+### Building
+
+The project is configured and built using the meson buildsystem, you will need to create a build directory and then configure the build depending on what you want.
+
+The project has a couple of build options you may configure, which affect the resulting binaries, this is how you will configure the firmware for you respective probe, or enable certain features.
+
+A non-exhaustive list of project options:
+ * probe: Hardware platform where the BMD firmware will run
+ * targets: Enabled debug targets
+ * debug_output: Enable debug output (for debugging the BMD stack, not debug targets)
+ * rtt_support: Enable RTT (Real Time Transfer) support
+
+You may see all available project options and valid values under  `Project options` in the output of the `meson configure` command.
+
+The first configuration commands are ran from the root of your clone of the blackmagic repository
+``` sh
+cd /path/to/repository/blackmagic
+```
+
+All following commands are ran from inside the build directory.
+
+#### Building Black Magic Debug Firmware
+
+To build the firmware you need to configure the probe hardware the firmware will run on, as well as a cross-compilation toolchain used.
+
+For convenience, a cross-file for each supported hardware probe is available which provides a sane default configuration for it.
+
+The build configuration command for the native probe may look like:
+``` sh
+meson setup build --cross-file cross-file/native.ini
+```
+
+Alternatively (for advanced users), if you wish to configure manually, for instance while writing support for a new probe, or a different toolchain, you can run:
+``` sh
+meson setup build --cross-file cross-file/arm-none-eabi.ini -Dprobe=native
+```
+
+You now should have a `build` directory from where you can build the firmware, this is also where your binary files will appear.
+
+The command `meson compile` will build the default targets:
+```sh
+cd build
+meson compile
+```
```

Comment:
No, as any possibly advanced (or clueless) user can say `--backend=vs2010` or `--backend=xcode` and this will not work. An alternative would be the oneliner `meson compile -C build`.

Reply:
Oh, true - it's only equivalent on a typical platform and with defaults for the cross-build.. hmm 🤔. We'll let Perigoso decide what they want to do here then and if they should ignore us or write some sort of note

Reply:
I chose this because we know it will always just work, even if slightly more verbose, any user comfortable with meson will know they can call ninja directly. It's also clear what the command does for any user, and means we are always interacting with same tool, instead of introducing another one which could possibly induce confusion.

Reply:
I chose this because we know it will always just work, even if slightly more verbose, any user comfortable with meson will know they can call ninja directly. It's also clear what the command does for any user, and means we are always interacting with same tool, instead of introducing another one which could possibly induce confusion.

Reply:
Yep, ok, we'll accept that rationale. Fair enough!